src/modules/alerting.ts

```diff
@@ -32,6 +26,42 @@ const pick = <
     return newObj;
 };
 
+type ToastOptions<Type extends ValidMessageType, Timeout extends number> = {
```

Comment:
This is working great @jakejrichards!  Optional props are back and like the enhanced descriptions in the type errors!